### PR TITLE
grass: update to 8.3.0

### PR DESCRIPTION
**Pull request by nilason**
#### Description

`grass`:
- Update to version 8.3.0
- Make the port `sqlite3` a "hardwired" dependant, remove the variant.
- Add Python 3.11 variant.

Rev-bump dependants `gdal-grass` and `qgis3` and  add new default Python 3.11 variant to QGIS, to keep in sync  with GDAL and GRASS.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 12.6.6 21G646 arm64
Xcode 14.2 14C18

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by nilason** [comment]
@Veence ping

File: gis/grass/Portfile [hunk 1]
```diff
@@ -71,8 +72,6 @@ configure.args-append \
     --with-pdal=${prefix}/bin/pdal-config \
     --with-tiff-includes=${prefix}/include \
     --with-tiff-libs=${prefix}/lib \
-    --with-png-includes=${prefix}/include \
-    --with-png-libs=${prefix}/lib \
```
Comment (herbygillot) [inline gis/grass/Portfile:75]: Do these specific changes remove any PNG functionality in this version of `grass`?

Comment (nilason) [reply gis/grass/Portfile:75]: No, just changes to configure, (see https://github.com/OSGeo/grass/pull/2679).

**Comment by nilason** [comment]
@herbygillot Thanks!

**Comment by herbygillot** [comment]
Np, thanks for your submission

**Pull request merged**